src/app/portlets/dot-edit-page/layout/dot-edit-layout-grid/dot-edit-layout-grid.component.ts

```diff
@@ -19,7 +20,8 @@ interface DotLayoutContainer {
     styleUrls: ['./dot-edit-layout-grid.component.scss']
 })
 export class DotEditLayoutGridComponent implements OnInit {
-    @Input() gridContainers: DotLayoutContainer[];
+    @Input() pageView: PageView;
```

Comment:
This component shouldn't care about pageView... he only care about grid item to render with the row and col information, in fact the tranformation of the model from pageView to the model that this component understand shouldn't be done here... move all that to the service, this component should only get the model as he expects.

Reply:
this component should not receive all the PageView because it dont need it, it should receive only the Layout.

but the tranformation must be done here, because it this componente who know about ng-grid

Reply:
Ok let's do something, lets create a service ```dot-edit-layout-grid.service.ts``` and he will be in charge of turning Layout into ng-grid and viceversa.

Reply:
it does need the PageView, since need to read the reference of the containers to make the transformation of the object.